drivers/opamp/opamp_stm32g4x_opamp.c

```diff
@@ -0,0 +1,666 @@
+/*
+ * Copyright (c) 2025 Alexander Kozhinov <ak.alexander.kozhinov@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/drivers/opamp.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+
+#include <stm32_ll_opamp.h>
+#include <stm32_ll_system.h>
+
+LOG_MODULE_REGISTER(opamp_stm32g4, CONFIG_OPAMP_LOG_LEVEL);
+
+#define DT_DRV_COMPAT st_stm32g4_opamp
+
+/* There is a spell-mistake in original driver - L268, stm32g4xx_ll_opamp.h */
+#if !defined(LL_OPAMP_INTERNAL_OUTPUT_DISABLED)
+/* checkpatch-ignore: TYPO_SPELLING */
+#define LL_OPAMP_INTERNAL_OUTPUT_DISABLED (0x00000000UL)
+#endif /* !LL_OPAMP_INTERNAL_OUTPUT_DISABLED */
+
+#if !defined(LL_OPAMP_INTERNAL_OUTPUT_ENABLED)
+/* checkpatch-ignore: TYPO_SPELLING */
+#define LL_OPAMP_INTERNAL_OUTPUT_ENABLED OPAMP_CSR_OPAMPINTEN
+#endif /* !LL_OPAMP_INTERNAL_OUTPUT_ENABLED */
+
+typedef enum {
+	OPAMP_INM_NONE = -1, /* used to represent states not present in ll */
+	/* LL_OPAMP_INPUT_INVERT_CONNECT_NO: only for OPAMP in follower or PGA
+	 * with positive gain without bias
+	 */
+	OPAMP_INM_NC = LL_OPAMP_INPUT_INVERT_CONNECT_NO,
+	OPAMP_INM_VINM0 = LL_OPAMP_INPUT_INVERT_IO0,
+	OPAMP_INM_VINM1 = LL_OPAMP_INPUT_INVERT_IO1,
+
+	OPAMP_INM_SEC_NONE = -2, /* used to represent states not present in ll */
+	/* only applicable in standalone mode */
+	OPAMP_INM_SEC_VINM0 = LL_OPAMP_INPUT_INVERT_IO0_SEC,
+	/* only applicable in standalone mode */
+	OPAMP_INM_SEC_VINM1 = LL_OPAMP_INPUT_INVERT_IO1_SEC,
+} opamp_inm_t;
```

Comment:
Don't define aliases for LL types. Defining dummy values is fine but use `#define`
```suggestion
#define OPAMP_INM_NONE -1
#define OPAMP_INM_SEC_NONE -2
```
I'm not sure there even is a need to distinguish `OPAMP_INM_NONE` from `OPAMP_INM_SEC_NONE` - a single `OPAMP_INPUT_NONE` would probably suffice